File: UWP/Shiba/ExtensionExecutors/ResExecutor.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Windows.UI.Xaml;
using Shiba.Controls;
using Shiba.ExtensionExecutors;

namespace WeiPo.Shiba.ExtensionExecutors
{
    public class ResExecutor : IMutableExtensionExecutor
    {
        public object ProvideValue(IShibaContext context, ShibaExtension value)
        {
            return Application.Current.Resources.ContainsKey(value.Value) ? Application.Current.Resources[value.Value] : null;
        }

        public string Name { get; } = "res";
    }
}

File: Android/app/src/main/java/moe/tlaster/weipo/shiba/extensionExecutors/ResExecutor.kt
package moe.tlaster.weipo.shiba.extensionExecutors

import android.graphics.Color
import moe.tlaster.shiba.IShibaContext
import moe.tlaster.shiba.extensionExecutor.IMutableExtensionExecutor
import moe.tlaster.shiba.type.ShibaExtension

class ResExecutor(override val name: String = "res") : IMutableExtensionExecutor {
    override fun provideValue(context: IShibaContext?, extension: ShibaExtension): Any? {
        return Color.TRANSPARENT
    }
}